shared/settings/invites/index.desktop.js

```diff
@@ -0,0 +1,150 @@
+// @flow
+import React from 'react'
+import moment from 'moment'
+import {globalStyles, globalColors, globalMargins} from '../../styles'
+import {Avatar, Box, Button, Divider, Icon, Input, Text} from '../../common-adapters'
+import {stateColors} from '../../util/tracker'
+import {intersperseFn} from '../../util/arrays'
+import SubHeading from '../subheading'
+
+import type {Props, PendingInvite, PendingEmailInvite, PendingURLInvite, AcceptedInvite} from './index'
+
+function intersperseDividers (arr) {
+  return intersperseFn(i => <Divider key={i} style={{backgroundColor: globalColors.black_05}} />, arr)
+}
+
+function Invites (props: Props) {
+  return (
+    <Box style={{...globalStyles.flexBoxColumn}}>
+      <Box style={{...globalStyles.flexBoxColumn, minHeight: 269, alignItems: 'stretch', marginTop: globalMargins.small}}>
+        <Input
+          floatingLabelText="Friend's email (optional)"
+          value={props.inviteEmail}
+          onChangeText={props.onChangeInviteEmail}
+          errorText={props.emailError ? ' ' : null}
+          style={{marginBottom: 0}}
+        />
+        {props.showMessageField && <Input
+          floatingLabelText='Message (optional)'
+          multiline={true}
+          value={props.inviteMessage}
+          onChangeText={props.onChangeInviteMessage}
+        />}
+        <Button
+          type='Primary'
+          label='Generate invitiation'
+          onClick={props.onGenerateInvitation}
+          style={{alignSelf: 'center', marginTop: globalMargins.medium}}
+        />
+      </Box>
+      {props.pendingInvites.length > 0 && <Box style={{...globalStyles.flexBoxColumn, marginBottom: 16}}>
+        <SubHeading>Pending invites ({props.pendingInvites.length})</SubHeading>
+        {intersperseDividers(props.pendingInvites.map(invite =>
+          <PendingInviteItem
+            key={invite.id}
+            invite={invite}
+            onReclaim={id => props.onReclaimInvitation(id)}
+          />
+        ))}
+      </Box>}
+      <Box style={globalStyles.flexBoxColumn}>
+        <SubHeading>Accepted invites ({props.acceptedInvites.length})</SubHeading>
+        {intersperseDividers(props.acceptedInvites.map(invite =>
+          <AcceptedInviteItem
+            key={invite.id}
+            invite={invite}
+            onClick={() => props.onClickUser(invite.username)}
+          />
+        ))}
+      </Box>
+    </Box>
+  )
+}
+
+function PendingInviteItem ({invite, onReclaim}: {invite: PendingInvite, onReclaim: (id: string) => void}) {
+  return (
+    <Box style={styleInviteItem}>
+      {invite.type === 'pending-email' ? <PendingEmailContent invite={invite} /> : <PendingURLContent invite={invite} />}
+      <Box style={{flex: 1}} />
```

Comment:
not to change anything, but we could also use space-between and not have this growing box